### ISSUE TITLE
using soot in java

### ISSUE THREAD

**Issue by Shabnam-t** [Sable/soot#1253]
hello I am new in soot and I have a little background in java. now I want using soot for analyzing java code I do not how can use it in java code?hoe use it in main class?

**Comment by ftc** [comment]
I would highly recommend reading "A Survivor's Guide to Java Program Analysis with Soot" which can be found at the following link:

https://www.brics.dk/SootGuide/

For a quick example, you can skip to section 4.3 " Extending Soot’s Main class".  The class `MyAnalysisTagger` is a subclass of [BodyTransfomer](https://github.com/Sable/soot/blob/4a04145d1a4d2713ef2733e33abd5ba77b2fd848/src/main/java/soot/BodyTransformer.java) that you implement.

**Comment by Shabnam-t** [comment]
Tank you so much.but for fed java to soot which instructions are necessary?

**Comment by ftc** [comment]
Typically I compile the java code to .class files or a .jar and run Soot on that. It is possible to run soot directly on the `.java` files but there are disadvantages to that.  This page in the wiki has more details:

https://github.com/Sable/soot/wiki/Running-Soot

The example from section 4.3 passes command-line options that it receives directly to Soot via the args variable.

**Comment by Shabnam-t** [comment]
I want compiling in eclipse.

**Comment by ftc** [comment]
Unfortunately, I don't use Eclipse so I can't help much there.  You may want to look through the wiki, there are a few articles involving the Eclipse ide.

**Comment by Shabnam-t** [comment]
Thank you but i can not find it

**Comment by anddann** [comment]
I totally agree, we should provide an example folder that shows how soot can be used and configured from Java code directly.
Do you mind sharing your code as a PR?

**Comment by Shabnam-t** [comment]
Yes exactly

**Comment by anddann** [comment]
super, can you reference this issue in your PR
